### PR TITLE
Add rke2 support to logs-collector

### PR DESCRIPTION
**Pull request by dkeightley**
Added rke2 detection and collection, small general updates for consistency

Updated wording for 'runtime' to 'distribution' as this seemed more relevant (k3s/rke2 both use containerd etc). Tests so far have been done on Ubuntu 20.04.

- [x] Test on RKE2 server/agent
- [x] Test on RKE control plane node
- [x] Test on RKE worker node
- [ ] Test on k3s server/agent

### CODE REVIEW

File: collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh [hunk 1]
```diff
@@ -63,35 +63,47 @@ sherlock() {
...
               FOUND+="k3s"
           fi
       fi
-      if [ -z $RUNTIME ]
+      if $(command -v rke2 >/dev/null 2>&1)
         then
-          echo -e "\n$(timestamp): couldn't detect container runtime"
+          RKE2_BIN="/var/lib/rancher/rke2/bin"
+          CRI_CONFIG_FILE="/var/lib/rancher/rke2/agent/etc/crictl.yaml"
+          if $(${RKE2_BIN}/crictl ps >/dev/null 2>&1)
```
Comment (dnoland1) [inline collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh:96]: Tested on RKE2 and this check seems to fail and not detect the cluster is rke2. Ran the crictl ps command manually and get this:
```
root@ip-10-0-7-10:~# /var/lib/rancher/rke2/bin/crictl ps
WARN[0000] runtime connect using default endpoints: [unix:///var/run/dockershim.sock unix:///run/containerd/containerd.sock unix:///run/crio/crio.sock]. As the default settings are now deprecated, you should set the endpoint instead.
ERRO[0002] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded
WARN[0002] image connect using default endpoints: [unix:///var/run/dockershim.sock unix:///run/containerd/containerd.sock unix:///run/crio/crio.sock]. As the default settings are now deprecated, you should set the endpoint instead.
ERRO[0004] connect endpoint 'unix:///var/run/dockershim.sock', make sure you are running as root and the endpoint has been started: context deadline exceeded
FATA[0004] listing containers: rpc error: code = Unimplemented desc = unknown service runtime.v1alpha2.RuntimeService
```
This VM does have docker installed already, so not sure if that's interfering.

Comment (dkeightley) [reply collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh:96]: Thanks, looks like `CRI_CONFIG_FILE` var needs to be inherited by subprocesses, I already had this exported on my nodes, added this to the script

**Review by dnoland1 — COMMENTED** [review]
Does it make sense to include the journald logs for rke2-server and rke2-agent?
```
# Included journald logs
JOURNALD_LOGS=(docker k3s containerd cloud-init systemd-network kubelet kubeproxy)
```

**Review by dnoland1 — APPROVED** [review]
LGTM - tested on a single Ubuntu rke2 server.

**Comment by dnoland1** [comment]
Tested on k3s as well.

**Pull request merged**